typescript/web/src/hooks/use-image-navigation.fixtures.ts

```diff
@@ -0,0 +1,81 @@
+import { GET_ALL_IMAGES_OF_A_DATASET_QUERY } from ".";
+import {
+  GetAllImagesOfADatasetQuery,
+  GetAllImagesOfADatasetQueryVariables,
+  GetAllImagesOfADatasetQuery_dataset_images,
+} from "../graphql-types/GetAllImagesOfADatasetQuery";
+import {
+  ApolloMockResponse,
+  ApolloMockResponses,
+} from "../utils/tests/apollo-mock";
+import { BASIC_DATASET_DATA } from "../utils/tests/data.fixtures";
+import { CURRENT_IMAGE } from "./test-utils";
+
+export const IMAGE1 = {
+  id: "929f5445-812c-4679-86bd-0f0748679f38",
+  url: "",
+  thumbnail200Url: "",
+};
+
+export const IMAGE2 = {
+  id: "51e00f79-d178-4ea1-8bb5-b0fa647f9d19",
+  url: "",
+  thumbnail200Url: "",
+};
+
+export const IMAGE3 = {
+  id: "8067a965-d017-4c4b-a82d-98cc803d1020",
+  url: "",
+  thumbnail200Url: "",
+};
+
+const CURRENT_IMAGE_IS_FIRST = [CURRENT_IMAGE, IMAGE1, IMAGE2];
+const CURRENT_IMAGE_IS_SECOND = [IMAGE1, CURRENT_IMAGE, IMAGE2];
+const CURRENT_IMAGE_IS_LAST = [IMAGE1, IMAGE2, CURRENT_IMAGE];
+const CURRENT_IMAGE_NOT_PRESENT = [IMAGE1, IMAGE2, IMAGE3];
+
+const getImagesMockGenerator = (
+  images: GetAllImagesOfADatasetQuery_dataset_images[]
+): ApolloMockResponse<
+  GetAllImagesOfADatasetQuery,
+  GetAllImagesOfADatasetQueryVariables
+> => {
+  return {
+    request: {
+      query: GET_ALL_IMAGES_OF_A_DATASET_QUERY,
+      variables: {
+        slug: BASIC_DATASET_DATA.slug,
+        workspaceSlug: BASIC_DATASET_DATA.workspace.slug,
+      },
+    },
+    result: {
+      data: {
+        dataset: {
+          id: BASIC_DATASET_DATA.id,
+          images,
+        },
+      },
+    },
+    nMatches: Number.POSITIVE_INFINITY,
```

Comment:
The mocked query is not a function. Why is that needed exactly?

Reply:
Are you talking about `nMatches` property ? If yes, I don't understand the `query is not a function` part

It's needed because the mutation is used in parallel. If you don't specify it, you will receive a warning saying that the mutation is not mocked after the first call

Reply:
Yes, I'm talking about `nMatches`. Restricting a query mock to a certain number of calls adds a layer of trust. It lets you be sure that the hook don't overcall the query or that the skip property is honored...

> I don't understand the query is not a function part

Yes, I was unclear. I meant "the >mocked< query >response< is not a function". Because, the most common case where you'd want to set nMatches to `Number.POSITIVE_INFINITY` is when you've implemented the mocked query return value as a function instead of static value.

To follow-up my question, why is `GET_ALL_IMAGES_OF_A_DATASET_QUERY` be called in parallel exactly?

Reply:
Actually what I said was wrong, this mock does not need this nMatches property. I was confused because of my 2 other PRs I guess 😄